cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -65,6 +66,54 @@ TriangleMesh::TriangleMesh(const core::Tensor &vertex_positions,
     SetVertexPositions(vertex_positions);
     SetTriangleIndices(triangle_indices);
 }
+    
+std::pair<core::Tensor, core::Tensor> TriangleMesh::ComputeAdjacencyList() {
+    
+    if (IsEmpty()) {
+        utility::LogWarning("TriangleMesh is empty. No attributes computed!");
+        return std::make_pair(core::Tensor(), core::Tensor());
+    }
+
+    if(!HasTriangleIndices()) {
+        utility::LogWarning("TriangleMesh has no Indices. No attributes computed!");
+        return std::make_pair(core::Tensor(), core::Tensor());
+    }
+
+    core::Tensor tris_cpu =
+            GetTriangleIndices().To(core::Device()).Contiguous();
+
+    std::unordered_map< size_t, std::set<size_t> > adjacencyList;
+    auto insertEdge = [&adjacencyList](size_t s, size_t t){
+        adjacencyList[s].insert(t);
+    };
```

Comment:
Edges are undirected. For a triangle mesh, 2 neighboring faces will share an edge. If they are oriented in-consistently, the directions will be opposite, i.e. if one face is [0, 1, 2], another may be [2, 1, 3]. In this case we will get duplicate entries in the adjacency list. Essentially, we just need the lower triangle of the adjacency matrix - the upper triangle is redundant.

Reply:
Also, some tri meshes are not manifold (degenerate) - this can lead to situations where an edge is shared by 3 faces. Using a set as the value type does address this.

PS: Why do we need an `unordered_map<size_t, set>` instead of a simpler `vector<set>`?

Reply:
yeah, vector would work. I will update